apps/web/src/views/BuyCrypto/hooks/usePriceQuoter.ts

```diff
@@ -68,49 +67,61 @@ const usePriceQuotes = (amount: string, inputCurrency: string, outputCurrency: s
   const [quotes, setQuotes] = useState<ProviderQoute[]>([])
   const { chainId } = useActiveChainId()
 
-  const fetchProviderAvailability = async (ip: string, combinedData: ProviderQoute[]) => {
-    // first check user availability
-    const responsePromises = [fetchMoonpayAvailability(ip), fetchMercuryoAvailability(ip)]
-    const responses = await Promise.allSettled(responsePromises)
+  const fetchProviderAvailability = useCallback(
+    async (ip: string, combinedData: ProviderQoute[]) => {
+      // first check user availability
+      const responsePromises = [fetchMoonpayAvailability(ip), fetchMercuryoAvailability(ip)]
+      const responses = await Promise.allSettled(responsePromises)
 
-    const dataPromises = responses.reduce((accumulator, response) => {
-      if (response.status === 'fulfilled') {
-        return [...accumulator, response.value]
-      }
-      console.error('Error fetching price quotes:', response.reason)
-      return accumulator
-    }, [])
+      const dataPromises = responses.reduce((accumulator, response) => {
+        if (response.status === 'fulfilled') {
+          return [...accumulator, response.value]
+        }
+        console.error('Error fetching price quotes:', response.reason)
+        return accumulator
+      }, [])
 
-    const [moonPayAvailability, mercuryoAvailability] = await Promise.all(dataPromises)
+      const [moonPayAvailability, mercuryoAvailability] = await Promise.all(dataPromises)
 
-    const ProviderAvailability: ProviderAvailabilityData = {
-      MoonPay: moonPayAvailability?.result.result.isAllowed ?? false,
-      Mercuryo: mercuryoAvailability?.result.result.data?.country.enabled ?? false,
-      BinanceConnect: true,
-    }
-    const sortedFilteredQuotes = combinedData.filter((quote: ProviderQoute) => {
-      return ProviderAvailability[quote.provider]
-    })
+      const ProviderAvailability: ProviderAvailabilityData = {
+        MoonPay: moonPayAvailability?.result.result.isAllowed ?? false,
+        Mercuryo: mercuryoAvailability?.result.result.data?.country.enabled ?? false,
+        BinanceConnect: true,
+      }
+      const sortedFilteredQuotes = combinedData.filter((quote: ProviderQoute) => {
+        return ProviderAvailability[quote.provider]
+      })
 
-    if (sortedFilteredQuotes.length > 1)
-      sortedFilteredQuotes.sort((a: ProviderQoute, b: ProviderQoute) => {
-        const totalAmountA = a.amount + a.providerFee + a.networkFee
-        const totalAmountB = b.amount + b.providerFee + b.networkFee
+      if (sortedFilteredQuotes.length > 1)
+        sortedFilteredQuotes.sort((a, b) => {
+          let totalAmountA = 0
+          let totalAmountB = 0
+
+          if (inputCurrency.toUpperCase() === 'ETH' || inputCurrency.toUpperCase() === 'BNB') {
```

Comment:
For a more less chain specific method, I think we better use the `NATIVE` from `@pancekswap/sdk`